test/core/gc.spec.js

```diff
@@ -0,0 +1,245 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const isNode = require('detect-node')
```

Comment:
i would suggest you use our ipfs-utils for this https://github.com/ipfs/js-ipfs-utils/blob/master/src/env.js#L15 it's safer

Reply:
It would be great to remove the `detect-node` dependency if this is available in `ipfs-utils` already.